### PR TITLE
add regression plots and error histograms for training/validation/testing

### PR DESCRIPTION
**Pull request by wiederm**
### Training progression plots

This PR adds regression plots and error histograms for
- each training/validation set evaluation
- final validation/test set evaluation
and logs these plots automatically to Weights&Biases (or tensorboard).

The error is plotted as a histogram with frequency on a log scale to identify outliers better:
image
The regression plot is a simple xy scatter plot:
image

During training we plot:
- energy/force error histogram
- energy regression

During validation we plot:
- energy error histogram
- energy regression

### Monitor outliers in the training and validation set

To better identify outliers, we log to W&B a counter for the k data points with the highest RMSE error. This allows us to match outliers to the histograms if needed.
image

## Pull Request Checklist
 - [x] Appropriate docstring(s) added/updated
 - [ ] Appropriate .rst doc file(s) added/updated
 - [x] PR is ready for review

### CODE REVIEW

**Comment by wiederm** [comment]
@chrisiacovella , can you plaese review?
